### PR TITLE
Adding publishPipelineMetadata to HelmDeploy

### PR DESCRIPTION
**Pull request by shigupt202**
**Task name**: HelmDeployV0

**Description**: Adding `publishMetadataFromTask` input to HelmDeploy to enable users to turn on/off publish metadata step

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) Y

**Attached related issue:** (Y/N) Y

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/HelmDeployV0/task.json [hunk 1]
```diff
@@ -434,6 +434,14 @@
             "helpMarkDown": "If this is true, this task will fail if any errors are written to the error pipeline, or if any data is written to the Standard Error stream. Otherwise the task will rely on the exit code to determine failure.",
             "groupName": "advanced"
         },
+        {
+            "name": "publishMetadataFromTask",
+            "type": "boolean",
+            "label": "Publish metadata from task",
```
Comment (ds-ms) [inline Tasks/HelmDeployV0/task.json:440]: publish pipeline metadata 

remove `fromTask`

**Review by ds-ms — APPROVED** [review]
Other than the name, everything else LGTM

File: Tasks/HelmDeployV0/task.json (hunk 1 above)
Comment (shigupt202) [reply Tasks/HelmDeployV0/task.json:440]: You mean the label or just the variable name?

Comment (ds-ms) [reply Tasks/HelmDeployV0/task.json:440]: Both

**Comment by keerthipriya54321** [comment]
Previously this used to be as a warning message. But, now it turned into ##[Error]Capturing deployment metadata failed with error: YAMLException:
Do anyone have idea why it turned into error from warning.

**Comment by jlangford9284** [comment]
I am also randomly having the same issue as @keerthipriya54321 

`##[error]Capturing deployment metadata failed with error: YAMLException: duplicated mapping key at line 176, column -73:
    metadata:
    ^`
    
I cannot reproduce this locally

**Pull request merged**